src/v/storage/segment_appender.cc

```diff
@@ -336,15 +336,23 @@ ss::future<> segment_appender::do_next_adaptive_fallocation() {
                  return _out.allocate(_fallocation_offset, step)
                    .then([this, step] { _fallocation_offset += step; });
              })
-      .handle_exception([this](std::exception_ptr e) {
-          vassert(
-            false,
+      .handle_exception_type([this](std::system_error e) {
+          vlog(
+            stlog.error,
```

Comment:
we still probably want to assert(false) on all non-enospc exceptions?

Reply:
Yep and I wanted to pick your brain on how you'd do this (C++/seastar-wise)